File: libs/gomodels/channel_role_permission.go
package model

type ChannelRolePermission struct {
	ID           string `gorm:"column:id;primary_key;type:uuid;default:uuid_generate_v4()"`
	RoleID       string `gorm:"column:roleId;type:uuid;"`
	PermissionID string `gorm:"column:permissionId;type:uuid;"`

	Role       *ChannelRole    `gorm:"foreignKey:RoleID"`
	Permission *RolePermission `gorm:"foreignKey:PermissionID"`
}

func (ChannelRolePermission) TableName() string {
	return "channels_roles_permissions"
}

File: libs/typeorm/src/entities/ChannelRolePermission.ts
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';

// eslint-disable-next-line import/no-cycle
import { ChannelRole } from './ChannelRole';
// eslint-disable-next-line import/no-cycle
import { RoleFlag } from './RoleFlag';

@Entity('channels_roles_permissions')
export class ChannelRolePermission {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  roleId: string;

  @Column()
  flagId: string;

  @ManyToOne(() => ChannelRole, _ => _.permissions, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'roleId' })
  role?: ChannelRole;

  @ManyToOne(() => RoleFlag, _ => _.channelRolePermissions)
  @JoinColumn({ name: 'flagId' })
  flag?: RoleFlag;
}